interview-questions.md

```diff
@@ -5,55 +5,65 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. What is a hash in Ruby?
+1. What is a hash in Ruby? 
 
   Your answer:
+  Ruby is an object orientated language. In other words, everything is an object even variables and data types like string, number, float they are all objects. A hash is used on those Ruby objects to inherit some properties of the object and class in Ruby. During my time as a developer, I have done some projects involving Ruby and also solved some Ruby programming challenges. Being able to manipulate objects in Ruby is important in order to solve problems and create objects that work with each other within your program. A hash is a special code that the computer uses to check and see if the item being entered has integrity in other words it's the same item as previously and it hasn't changed in any way. 
 
   Researched answer:
+  My answer was partially correct. I left out a key aspect of it being a data structure which contains a set of objects called keys. The hash refers to both the key and its value. Another name for a hash is associative array. It's called that because it's job is to associate value with each of the keys. It doesn't work like an array because it uses the object not an integer like an array does. You can create a hash using class method and the keyword new. Syntax is restaurant = Hash.new this creates an empty hash because it doesn't have any keyword associated with it. Hashes can be altered by adding or deleting a key value/pair in a hash. The key can also be changed.
 
```

Comment:
Awesome answers. When you stated `During my time as a developer, I have done some projects involving Ruby and also solved some Ruby programming challenges.` Can you give me an example?